paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/PaymentMethodService.java

```diff
@@ -0,0 +1,232 @@
+            String filterKey = getFilterKeys(paymentMethodFilter, request);
+
+            HashMap<String, List<String>> filterValues = getFilterValuesMap(paymentMethodFilter);
+
+            for (Map.Entry<String, List<String>> entry : filterValues.entrySet()) {
+                String key = entry.getKey();
+
+                if (filterKey.equalsIgnoreCase(key)) {
+                    List<String> paymentMethodGroups = entry.getValue();
+
+                    for (String value : paymentMethodGroups) {
+                        filteredMethodsList.addAll(Arrays.stream(methods)
+                                .filter(method -> method.getGroup().equalsIgnoreCase(value))
+                                .collect(Collectors.toList()));
+                    }
+                }
+            }
+        }
+
+        if (!filteredMethodsList.isEmpty()) {
+            methods = filteredMethodsList.toArray(new PaymentMethodDto[0]);
+        }
+
+        return methods;
+    }
+
+    private List<PaymentMethodFilter> getFiltersEnabledForNamespace(String namespace) {
+        List<PaymentMethodFilter> filtersForNamespace = new ArrayList<>();
+
+        for (PaymentMethodFilter methodFilter : PaymentMethodFilter.getAll()) {
+            String orderTypeFilterEnabledList = env.getRequiredProperty("enabled_namespaces.payment_method_filter." + methodFilter);
+
+            if (orderTypeFilterEnabledList.contains(namespace)) {
+                filtersForNamespace.add(methodFilter);
+            }
+        }
+
+        return filtersForNamespace;
+    }
+
+    public String getFilterKeys(PaymentMethodFilter filter, GetPaymentMethodListRequest request) {
+        OrderDto orderDto = request.getOrderDto();
+
+        if (filter.equals(PaymentMethodFilter.ORDER_TYPE)) {
+            return orderDto.getType();
+        }
+
+        throw new CommonApiException(
+                HttpStatus.INTERNAL_SERVER_ERROR,
+                new Error("unknown-payment-method-filter-type", "unknown payment method filter type")
+        );
+    }
+
+    // namespace can be used as a parameter if different namespaces need to return different values for filter type
+    public HashMap<String, List<String>> getFilterValuesMap(PaymentMethodFilter filter) {
+        HashMap<String, List<String>> values = new HashMap<>();
+
+        if (filter.equals(PaymentMethodFilter.ORDER_TYPE)) {
+            values.put(OrderType.SUBSCRIPTION, Collections.singletonList(PaymentType.CREDIT_CARDS));
+        }
+
+        return values;
+    }
+
+    public List<PaymentMethodDto> getAllPaymentMethods() {
+        Iterable<PaymentMethodModel> paymentMethods = paymentMethodRepository.findAll();
+
+        List<PaymentMethodDto> paymentMethodDtos = StreamSupport.stream(paymentMethods.spliterator(), false)
+                .map(paymentMethodMapper::toDto)
+                .collect(Collectors.toList());
+        return paymentMethodDtos;
```

Comment:
You can inline this variable because it is reduntant

Reply:
Very true, I will do the changes.